core/backend/src/annotations/TextAnnotationElement.ts

```diff
@@ -549,14 +549,19 @@ function updateTextBlocks(elem: TextAnnotation2d | TextAnnotation3d, textBlocks:
  * Uses the same semantics as [ECVersion]($ecschema-metadata).
  * @internal
 */
-export const TEXT_STYLE_SETTINGS_JSON_VERSION = "1.0.0";
+// 1.0.1 - Added terminatorShapes for leaders
+export const TEXT_STYLE_SETTINGS_JSON_VERSION = "1.0.1";
 
 function migrateTextStyleSettings(oldData: VersionedJSON<TextStyleSettingsProps>): TextStyleSettingsProps {
   if (oldData.version === TEXT_STYLE_SETTINGS_JSON_VERSION) return oldData.data;
 
-  // Place migration logic here.
+  if (oldData.data.leader && !oldData.data.leader.terminatorShape) {
```

Comment:
id recommend adding a comment above to note that this is the migration needed for 1.0.0 to 1.0.1